packages/angular_devkit/build_angular/src/builders/dev-server/tests/behavior/serve_service-worker_spec.ts

```diff
@@ -37,182 +41,187 @@ describeServeBuilder(
+                  ignoreVary: true,
                 },
-              ],
-              dataGroups: [],
-              hashTable: {
-                '/favicon.ico': '84161b857f5c547e3699ddfbffc6d8d737542e01',
-                '/assets/folder-asset.txt': '617f202968a6a81050aa617c2e28e1dca11ce8d4',
-                '/index.html': '9d232e3e13b4605d197037224a2a6303dd337480',
-                '/spectrum.png': '8d048ece46c0f3af4b598a95fd8e4709b631c3c0',
+                patterns: [],
               },
-            }),
-          );
+            ],
+            dataGroups: [],
+            hashTable: {
+              '/favicon.ico': '84161b857f5c547e3699ddfbffc6d8d737542e01',
+              '/assets/folder-asset.txt': '617f202968a6a81050aa617c2e28e1dca11ce8d4',
+              '/index.html': isViteRun
+                ? 'e5b73e6798d2782bf59dd5272d254d5bde364695'
+                : '9d232e3e13b4605d197037224a2a6303dd337480',
+              '/media/spectrum.png': '8d048ece46c0f3af4b598a95fd8e4709b631c3c0',
+            },
+          }),
+        );
+      });
+
+      it('works with localize', async () => {
+        setupTarget(harness, {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          serviceWorker: (isViteRun ? 'ngsw-config.json' : true) as any,
+          resourcesOutputPath: isViteRun ? undefined : 'media',
+          assets: ['src/favicon.ico', 'src/assets'],
+          styles: ['src/styles.css'],
+          localize: ['fr'],
         });
 
-        it('works with localize', async () => {
-          setupTarget(harness, {
-            serviceWorker: true,
-            assets: ['src/favicon.ico', 'src/assets'],
-            styles: ['src/styles.css'],
-            localize: ['fr'],
-          });
+        await harness.writeFiles({
+          'ngsw-config.json': JSON.stringify(manifest),
+          'src/assets/folder-asset.txt': 'folder-asset.txt',
+          'src/styles.css': `body { background: url(./spectrum.png); }`,
+        });
 
-          await harness.writeFiles({
-            'ngsw-config.json': JSON.stringify(manifest),
-            'src/assets/folder-asset.txt': 'folder-asset.txt',
-            'src/styles.css': `body { background: url(./spectrum.png); }`,
-          });
+        harness.useTarget('serve', {
+          ...BASE_OPTIONS,
+        });
 
-          harness.useTarget('serve', {
-            ...BASE_OPTIONS,
-          });
+        const { result, response } = await executeOnceAndFetch(harness, '/ngsw.json');
 
-          const { result, response } = await executeOnceAndFetch(harness, '/ngsw.json');
+        expect(result?.success).toBeTrue();
 
-          expect(result?.success).toBeTrue();
+        expect(await response?.json()).toBeDefined();
+      });
 
-          expect(await response?.json()).toBeDefined();
+      // TODO(fix-vite): currently this is broken in vite due to watcher never terminates.
+      (isViteRun ? xit : it)('works in watch mode', async () => {
```

Comment:
This conditional shouldn't be needed anymore.

Reply:
This test still fails